### ISSUE TITLE
User with auditor role gets email notification of the comments on the expense report and with wrong subject line

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#38600]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 
**Reproducible in staging?:** needs reproduction (not getting emails as I am using alias email)
**Reproducible in production?:** needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @heyjennahay 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1710776811985599

## Action Performed:
Precondition : Create a collect workspace and add employee, auditor and approve in OldDot.

1. Sign in as employee and create an expense report
2. Submit it to the approver
3. As a approver reject the report in OldDot with a comment
4. As a employee leave a comment in the transaction thread

## Expected Result:
Auditor shouldn't get any email notification since not tagged or mentioned in the comment

## Actual Result:
Auditor gets email notification with wrong subject line and poor formatting

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence
![image (11)](https://github.com/Expensify/App/assets/38435837/60dcab52-f342-40f8-bbd9-74ea0fa7ac2b)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013a56bd1987bbfab2</li>
        <li>Upwork Job ID: 1772842001198821376</li>
        <li>Last Price Increase: 2024-03-27</li>
    </ul>
</details>

**Comment by heyjennahay** [comment]
https://www.expensify.com/report?param={%22pageReportID%22:%223869299477758669%22,%22keepCollection%22:true}

**Comment by JmillsExpensify** [comment]
Oh so probably a backend bug, yeah? Adding the internal label as a result.

**Comment by JmillsExpensify** [comment]
Still waiting on an internal engineer to pick this up.

**Comment by JmillsExpensify** [comment]
Same

**Comment by JmillsExpensify** [comment]
Same

**Comment by JmillsExpensify** [comment]
Moving this bug to monthly, since it's not a priority or on the roadmap.

**Comment by JmillsExpensify** [comment]
Still waiting on internal engineer to prioritize

**Comment by JmillsExpensify** [comment]
I'm not seeing this issue get picked up internally, so it's clearly not a priority. We can re-open when someone is able to assign themselves and fix the underlying bug.
